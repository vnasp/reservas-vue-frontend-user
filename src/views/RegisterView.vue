<template>
  <main class="container pt-5 h-100 d-flex flex-column align-items-center">
    <h1 class="my-5 text-center"><i class="bi bi-bookmark-star-fill me-1"></i>Crea tu cuenta</h1>
    <p class="text-center w-75">Registrándote en Hostal Araucaria podrás reservar una habitación para ti y tus acompañantes, y disfrutar de una estadía inolvidable con servicios excepcionales y actividades emocionantes.</p>
    <form @submit.prevent="handleRegister" class="w-75">
      <div class="mb-3">
        <label for="nombre" class="form-label">Nombre</label>
        <input type="text" v-model="usuario.nombre" class="form-control" id="nombre" placeholder="Ingresa tu nombre" required>
      </div>
      <div class="mb-3">
        <label for="apellido" class="form-label">Apellido</label>
        <input type="text" v-model="usuario.apellido" class="form-control" id="apellido" placeholder="Ingresa tu apellido" required>
      </div>
      <div class="mb-3">
        <label for="email" class="form-label">Correo</label>
        <input type="email" v-model="usuario.email" class="form-control" id="email" placeholder="Ingresa tu e-mail" required>
      </div>
      <div class="mb-3">
        <label for="telefono" class="form-label">Teléfono</label>
        <input type="tel" v-model="usuario.telefono" class="form-control" id="telefono" placeholder="Ingresa tu teléfono" required>
      </div>
      <div class="mb-3">
        <label for="password" class="form-label">Contraseña</label>
        <input type="password" v-model="usuario.password" class="form-control" id="password" placeholder="Ingresa tu Contraseña" required>
      </div>
      <input type="hidden" v-model="usuario.adminStatus" value="false">
      <div class="text-center">
        <button type="submit" class="btn btn-primary">Registrarse</button>
      </div>
    </form>
  </main>
</template>

<script>
import { mapActions } from 'vuex';
import { v4 as uuidv4 } from 'uuid';

export default {
  name: 'RegisterView',
  data() {
    return {
      usuario: {
        id: uuidv4(),
        nombre: 'Usuario',
        apellido: 'Prueba',
        email: 'usuario@correo.com',
        telefono: '912345678',
        password: '1234',
        adminStatus: false
      }
    };
  },
  methods: {
    ...mapActions(['registrarUsuario']),
    handleRegister() {
      this.registrarUsuario(this.usuario);
      this.$router.push('/login');
    }
  },
}
</script>
