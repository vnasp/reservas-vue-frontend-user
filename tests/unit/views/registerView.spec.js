import { shallowMount } from '@vue/test-utils';
import RegisterView from '@/views/RegisterView.vue';

describe('RegisterView.vue', () => {
  it('Crear un snapshot de la vista de Registro', () => {
    // Arrange: Configuración inicial
    const wrapper = shallowMount(RegisterView);

    // Act: No aplica

    // Assert: Comprobación de que el HTML coincide con el snapshot
    expect(wrapper.html()).toMatchSnapshot();
  });
});