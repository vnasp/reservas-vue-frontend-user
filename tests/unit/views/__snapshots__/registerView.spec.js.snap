// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterView.vue Crear un snapshot de la vista de Registro 1`] = `
<main class="container pt-5 h-100 d-flex flex-column align-items-center">
  <h1 class="my-5 text-center"><i class="bi bi-bookmark-star-fill me-1"></i>Crea tu cuenta</h1>
  <p class="text-center w-75">Registrándote en Jack Rabbit Slim's podrás reservar una mesa para cuantos invitados quieras y disfrutar de un momento y platos inolvidables.</p>
  <form class="w-75">
    <div class="mb-3"><label for="nombre" class="form-label">Nombre</label><input type="text" class="form-control" id="nombre" placeholder="Ingresa tu nombre" required=""></div>
    <div class="mb-3"><label for="apellido" class="form-label">Apellido</label><input type="text" class="form-control" id="apellido" placeholder="Ingresa tu apellido" required=""></div>
    <div class="mb-3"><label for="email" class="form-label">Correo</label><input type="email" class="form-control" id="email" placeholder="Ingresa tu e-mail" required=""></div>
    <div class="mb-3"><label for="telefono" class="form-label">Teléfono</label><input type="tel" class="form-control" id="telefono" placeholder="Ingresa tu teléfono" required=""></div>
    <div class="mb-3"><label for="password" class="form-label">Contraseña</label><input type="password" class="form-control" id="password" placeholder="Ingresa tu Contraseña" required=""></div><input type="hidden" value="false">
    <div class="text-center"><button type="submit" class="btn btn-primary">Registrarse</button></div>
  </form>
</main>
`;
